contracts/chain-adapters/Ovm_L2_Forwarder.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { Arbitrum_L2_Forwarder, ITokenMessenger } from "./Arbitrum_L2_Forwarder.sol";
+import { ArbitrumInboxLike, ArbitrumL1ERC20GatewayLike } from "../interfaces/ArbitrumBridgeInterfaces.sol";
+import { LibOptimismUpgradeable } from "@openzeppelin/contracts-upgradeable/crosschain/optimism/LibOptimismUpgradeable.sol";
+import { Lib_PredeployAddresses } from "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @notice Contract containing logic to send messages from an OVM L2 to an Arbitrum-like L3.
+ * @dev This contract is very similar to Arbitrum_CustomGasToken_Adapter. It is meant to bridge
+ * tokens and send messages over a bridge which uses a custom gas token, except this contract makes
+ * the assumption that it is deployed on an OpStack L2.
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Ovm_L2_Forwarder is Arbitrum_L2_Forwarder {
```

Comment:
I keep getting lost in these naming conventions (i.e. L2 forwarder, l3 adapter) and it gets a bit tricker when chain names are overlaid like this. Is there a concrete naming scheme to be followed? wdyt about building a lexicon that explains how to interpret the names?

Reply:
I think we should stick with the <chainId>_Adapter_{base} naming scheme:
- chainId makes it clear which chain the contract should be deployed on, or L1 if omitted
- Adapter already is intuitively understood by us as something that sends message to a higher level
- Base contracts are meant to be inherited